include/mbedtls/config.h

```diff
@@ -56,6 +56,31 @@
 #define MBEDTLS_HAVE_ASM
 
 /**
+ * \def MBEDTLS_NO_UDBL_DIVISION
+ *
+ * The platform lacks support for double-width integer division (64-bit
+ * division on a 32-bit platform, 128-bit division on a 64-bit platform).
+ *
+ * Used in:
+ *      include/mbedtls/bignum.h
+ *      library/bignum.c
+ *
+ * The bignum code uses double-width division to speed up some operations.
+ * Double-width division is often implemented in software that needs to
+ * be linked with the program. The presence of a double-width integer
+ * type is usually detected automatically through preprocessor macros,
+ * but the automatic detection cannot know whether the code needs to
+ * and can be linked with an implementation of division for that type.
+ * By default division is assumed to be usable if the type is present.
+ * Uncomment this option to prevent the use of double-width division.
+ *
+ * Note that division for the native integer type is always required.
+ * Furthermore, a 64-bit type is always required even on a 32-bit
+ * platform, but it need not support multiplication or division.
+ */
```

Comment:
@hanno-arm According to the C standard (any version), if an arithmetic type exists then all the arithmetic operations must be supported. However, in practice, in some environments, division (and sometimes multiplication) requires a non-trivial implementation which involves the runtime library and not just the compiler. Technically, combining a compiler that provides a `uint128_t` type with a runtime library that doesn't provide 128-bit division/multiplication results in a non-compliant C implementation. And a runtime library that doesn't provide 64-bit division/multiplication is not C99-compliant.

However such environments are relatively common in the embedded world and it is reasonable to expect mbed TLS to support them.